### ISSUE TITLE
Differences between objectLanguage, workingLanguage, sourceLanguage, and targetLanguage

### ISSUE THREAD

**Issue by daliboris** [DARIAH-ERIC/lexicalresources#158]
I'm curious how should I use these roles for dictionary description. Can we have some examples?

Does the selection of the role depend on the kind of dictionary, for example, a modern monolingual dictionary contains the same objectLanguage and workingLanguage?

And what about bilingual dictionary? Should I use objectLanguage for headwords, workingLanguage for metalanguage (for example in Latin), sourceLanguage for language of the citations (will be identical with workingLanguage) and targetLanguage for the language of translation?

Can you help me, which role to choose?

**Comment by michmech** [comment]
The way I understand and use this terminology is as follows.

In a bilingual dictionary, the **source language** is the language of the headwords and the **target language** is the language of the translations.

If the bilingual language is an **encoding dictionary** (eg. a Czech-to-German dictionary for Czech-speaking learners of German) then the target language is the **object language** and the source language is the **working language** (I prefer to call it the metalanguage).

If the bilingual dictionary is a **decoding dictionary** (eg. a German-to-Czech dictionary for Czech-speaking learners of German) then it's the other way around: the source language is the **object language** and the target language is the **working language** (or metalanguage).

In other words, the object language is the language the dictionary is _about_, the language whose lexicon the dictionary _explains_, while the working language/metalanguage is the language in which the explanations are given.

All this terminology has its roots in the [European Dictionary Portal](http://www.dictionaryportal.eu/), a website I co-created back in 2016 as part of the [European Network of e-Lexicography](https://www.elexicography.eu/) (ENeL) COST Action.

**Comment by ttasovac** [comment]
I'm closing this old issue. There is now also an explanation in the specification of `language` and the `@role` attribute: https://dariah-eric.github.io/lexicalresources/pages/TEILex0/spec.html#TEI.language
